### ISSUE TITLE
add travis

### ISSUE THREAD

**Issue by chessai** [andrewthad/quickcheck-classes#113]
with haskell-ci, travis maintenance is a lot easier. i would do this myself, but i'm not the owner of this repo, so i can't have travis watch it, even with push access. i can pr a .travis.yml generated by haskell-ci; but you (@andrewthad) would have to add it to travis. this would make CI a lot easier for us, so we don't have to run a local build script. it would also check that things like haddocks build, cabal check passes, etc.

**Comment by chessai** [comment]
p.s. i am fine with maintaining the travis setup.

**Comment by andrewthad** [comment]
Yeah, I'm cool with having travis enabled. That would be pretty useful for this library.

**Comment by chessai** [comment]
travis is not going to be an option because they updated their pricing. we will have to use GitHub CI instead. which is fine because it's much better and still free
